rpc/src/v1/tests/integration/eth.rs

```diff
@@ -0,0 +1,227 @@
+		let req_new_acc = r#"{
+			"jsonrpc": "2.0",
+			"method": "eth_getBalance",
+			"params": ["0xaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"],
+			"id": 3
+		}"#;
+
+		let res_new_acc = r#"{"jsonrpc":"2.0","result":"0x00","id":3}"#.to_owned();
+		assert_eq!(tester.handler.handle_request(req_new_acc).unwrap(), res_new_acc);
+	});
+}
+
+#[test]
+fn eth_block_number() {
+	let chain = extract_chain!("BlockchainTests/bcRPC_API_Test");
+	chain_harness(chain, |tester| {
+		let req_number = r#"{
+			"jsonrpc": "2.0",
+			"method": "eth_blockNumber",
+			"params": [],
+			"id": 1
+		}"#;
+
+		let res_number = r#"{"jsonrpc":"2.0","result":"0x20","id":1}"#.to_owned();
+		assert_eq!(tester.handler.handle_request(req_number).unwrap(), res_number);
+	});
+}
+
+#[cfg(test)]
+#[test]
+fn eth_transaction_count() {
+	let chain = extract_chain!("BlockchainTests/bcRPC_API_Test");
+	chain_harness(chain, |tester| {
+		let address = tester.accounts.new_account("123").unwrap();
+
+		let req_before = r#"{
+			"jsonrpc": "2.0",
+			"method": "eth_getTransactionCount",
+			"params": [""#.to_owned() + format!("0x{:?}", address).as_ref() + r#"", "latest"],
+			"id": 15
+		}"#;
+
+		let res_before = r#"{"jsonrpc":"2.0","result":"0x00","id":15}"#;
+
+
+		assert_eq!(tester.handler.handle_request(&req_before).unwrap(), res_before);
+
+		let t = Transaction {
+			nonce: U256::zero(),
+			gas_price: U256::from(0x9184e72a000u64),
+			gas: U256::from(0x76c0),
+			action: Action::Call(Address::from_str("d46e8dd67c5d32be8058bb8eb970870f07244567").unwrap()),
+			value: U256::from(0x9184e72au64),
+			data: vec![]
+		}.fake_sign(address.clone());
+
+		let req_send_trans = r#"{
+			"jsonrpc": "2.0",
+			"method": "eth_sendTransaction",
+			"params": [{
+				"from": ""#.to_owned() + format!("0x{:?}", address).as_ref() + r#"",
+				"to": "0xd46e8dd67c5d32be8058bb8eb970870f07244567",
+				"gas": "0x76c0",
+				"gasPrice": "0x9184e72a000",
+				"value": "0x9184e72a"
+			}],
+			"id": 16
+		}"#;
+
+		// dispatch the transaction.
+		tester.handler.handle_request(&req_send_trans).unwrap();
+
+		// we have submitted the transaction -- but this shouldn't be reflected in a "latest" query.
+		let req_after_latest = r#"{
+			"jsonrpc": "2.0",
+			"method": "eth_getTransactionCount",
+			"params": [""#.to_owned() + format!("0x{:?}", address).as_ref() + r#"", "latest"],
+			"id": 17
+		}"#;
+
+		let res_after_latest = r#"{"jsonrpc":"2.0","result":"0x00","id":17}"#;
+
+		assert_eq!(&tester.handler.handle_request(&req_after_latest).unwrap(), res_after_latest);
+
+		// the pending transactions should have been updated.
+		let req_after_pending = r#"{
+			"jsonrpc": "2.0",
+			"method": "eth_getTransactionCount",
+			"params": [""#.to_owned() + format!("0x{:?}", address).as_ref() + r#"", "pending"],
+			"id": 18
+		}"#;
+
+		let res_after_pending = r#"{"jsonrpc":"2.0","result":"0x01","id":18}"#;
+
+		assert_eq!(&tester.handler.handle_request(&req_after_pending).unwrap(), res_after_pending);
```

Comment:
Right now, this really only tests the TestMinerService's ability to update transactions.
It would probably be useful to integrate this with the actual Miner, although I worry that this could lead to race conditions in tests like these, where we want to send the "pending" RPC request after the miner has placed the transaction in a pending block, but before the block is sealed.